src/contracts/core/PaymentCoordinator.sol

```diff
@@ -0,0 +1,490 @@
+        for (uint256 i = 0; i < rangePayments.length; i++) {
+            RangePayment calldata rangePayment = rangePayments[i];
+            bytes32 rangePaymentHash = keccak256(abi.encode(msg.sender, paymentNonce, rangePayment));
+            require(
+                !isRangePaymentHash[msg.sender][rangePaymentHash],
+                "PaymentCoordinator._payForRange: range payment hash already submitted"
+            );
+            isRangePaymentHash[msg.sender][rangePaymentHash] = true;
+
+            _payForRange(rangePayment);
+            emit RangePaymentCreated(msg.sender, paymentNonce, rangePaymentHash, rangePayment);
+            paymentNonce++;
+        }
+    }
+
+    /**
+     * @notice similar to `payForRange` except the payment is split amongst *all* stakers
+     * rather than just those delegated to operators who are registered to a single avs
+     * @param rangePayments The range payments being created
+     */
+    function payAllForRange(
+        RangePayment[] calldata rangePayments
+    ) external onlyWhenNotPaused(PAUSED_PAY_ALL_FOR_RANGE) onlyPayAllForRangeSubmitter nonReentrant {
+        for (uint256 i = 0; i < rangePayments.length; i++) {
+            RangePayment calldata rangePayment = rangePayments[i];
+            bytes32 rangePaymentForAllHash = keccak256(abi.encode(msg.sender, paymentNonce, rangePayment));
+            require(
+                !isRangePaymentForAllHash[msg.sender][rangePaymentForAllHash],
+                "PaymentCoordinator._payForRange: range payment hash already submitted"
+            );
+            isRangePaymentForAllHash[msg.sender][rangePaymentForAllHash] = true;
+
+            _payForRange(rangePayment);
+            emit RangePaymentForAllCreated(msg.sender, paymentNonce, rangePaymentForAllHash, rangePayment);
+            paymentNonce++;
+        }
+    }
+
+    /**
+     * @notice Claim payments against a given root (read from distributionRoots[claim.rootIndex])
+     * @param claim The PaymentMerkleClaim to be processed.
+     * Contains the root index, earner, payment leaves, and required proofs
+     * @dev only callable by the valid claimer, that is
+     * if claimerFor[claim.earner] is address(0) then only the earner can claim, otherwise only
+     * claimerFor[claim.earner] can claim the payments.
+     */
+    function processClaim(PaymentMerkleClaim calldata claim) external onlyWhenNotPaused(PAUSED_CLAIM_PAYMENTS) {
+        _processClaim(claim);
+    }
+
+    /**
+     * @notice Creates a new distribution root
+     * @param root The merkle root of the distribution
+     * @param paymentCalculationEndTimestamp The timestamp until which payments have been calculated
+     * @param activatedAt timestamp at which that the root can be claimed against
+     * @dev Only callable by the paymentUpdater
+     */
+    function submitRoot(
+        bytes32 root,
+        uint64 paymentCalculationEndTimestamp,
+        uint64 activatedAt
```

Comment:
remove

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/512/commits/b02c588081a4ce25e1f683513218acfb7e25f4ca#diff-2a28246efa54c09dff5f0afef55f490d130b1a13c5b7e30efb51d5647c52a1a2L168

Removed the activatedAt